docs/reference/api/connectivity/bluetooth/GattClient.md

```diff
@@ -1,6 +1,26 @@
 ## GattClient
 
-[Add description here.]
+GattClient defines procedures required for interacting with a remote GattServer.
+
+#### Discovery procedures
+
+A GattServer hosts a fixed set of services. These services are a logical composition of characteristics that may be discovered, read, or written, and can broadcast their state to a connected client. These characteristics may also contain metainformation and named characteristic descriptors. A characteristic descriptor may indicate the unit used for a characteristic value, describe the characteristic purpose in a textual form or allow a client to register for update notifications for the characteristic value.
+
+Prior to any interaction with a server characteristic, a GattClient discovers the layout of the services and characteristics present on the server.
+
+The layout of the descriptors of a characteristic may also be issued as an extra discovery step.
+
+#### Attribute manipulation
+
+As a result of the discovery process, the client can start interacting with the characteristic discovered. Depending on the characteristic properties (acquired during discovery), a client can read or write the value of a given characteristic.
+
+Mbed BLE abstracts most read and write operations to offer a single API that can be used to read or write characteristics values. Application code does not have to handle the fragmentation/reassembly process necessary if the attribute value to transported cannot fit in a single data packet.
```

Comment:
I don't understand what 'most' is doing in this 1st sentence (so deleted).  'read and write' is repetitive, but I am also confused about the overall meaning of the sentence.  I'm assuming you means that the abstracts read/write operations, and that these operations offer a single API?

2nd sentence: "attribute value to transported" something is wrong here.  Is it the attribute value to be transported (that will be transported)?